backend/hdf5/h5x/H5DataType.cpp

```diff
@@ -281,6 +281,12 @@ data_type_from_h5(H5T_class_t vclass, size_t vsize, H5T_sign_t vsign)
     return DataType::Nothing;
 }
 
+void check_bool_enum(h5x::DataType type) {
+    assert(type.member_count() == 2);
```

Comment:
`Assert` is to find programming errors. In general, if we didn't fuck up, then there should never be a case where any of the asserts are it. If for whatever reason that happens then in `data_type_from_h5` we return a `DataType::Nothing`, which is one of the sensible things to do. Throwing an exception would also work, but I guess the user should never actually see it.
In the bool case if the asserts are "not working" then we would accept something not a bool as a bool which we really shouldn't do.